### PR TITLE
CPLP-616: Add validation for user overlay input fields

### PR DESCRIPTION
**Pull request by christianraysono**
Adds useFormValidation hook
Implements validation for user overlay input fields
Renames tab state variables

### CODE REVIEW

**Comment by floriansommerraysono** [comment]
Thanks for renaming tab state variables. I was not happy with the value-value. :-)

**Review by floriansommerraysono — APPROVED** [review]

File: cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+    lastname: {
+      required: true,
+      pattern: /^.([A-Za-zÀ-ÿ-,.']{1,40}[ ]?){1,8}$/i,
+    },
+    email: {
+      required: true,
+      pattern:
+        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+    },
+  })
```
Comment (oyo) [inline cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx:22]: Do we really need the `required` attribute? We could specify that by giving a `pattern` that allows an empty string. Also I would prefer to combine that with the `userInputs` so we have all the configuration in one place like:
```
{
      key: 'firstname',
      label: t('global.field.first'),
      placeholder: t('global.field.first'),
      helperText: '',
      pattern: /^([A-Za-zÀ-ÿ-,.']{1,40}[ ]?){1,8}$/i,
},
```

File: cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
 import { Input } from 'cx-portal-shared-components'
 import { Box } from '@mui/material'
+import { useFormValidation } from 'hooks/useFormValidation'
 
 export const SingleUserContent = () => {
   const { t } = useTranslation()
+  const { handleChange, errors } = useFormValidation({
+    firstname: {
+      required: true,
+      pattern: /^.([A-Za-zÀ-ÿ-,.']{1,40}[ ]?){1,8}$/i,
```
Comment (oyo) [inline cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx:11]: why is that dot allowed as first character in `/^.(` ?

File: cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+    },
+    lastname: {
+      required: true,
+      pattern: /^.([A-Za-zÀ-ÿ-,.']{1,40}[ ]?){1,8}$/i,
```
Comment (oyo) [inline cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx:15]: first char arbitrary

**Review by oyo — CHANGES_REQUESTED** [review]
generally really good, but see my other comments

File: cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx (hunk 1 above)
Comment (christianraysono) [reply cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx]: The validation hook is now refactored to use the `userInputs` array.

File: cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx (hunk 2 above)
Comment (christianraysono) [reply cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx]: Pattern is fixed

File: cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx (hunk 3 above)
Comment (christianraysono) [reply cx-portal/src/components/pages/UserManagement/AddUserOverlay/SingleUserContent.tsx]: Pattern is fixed

**Review by oyo — APPROVED** [review]
looks good now

**Pull request merged**